includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -265,9 +258,10 @@ protected function update_post_meta( &$order ) {
 			update_post_meta( $id, '_shipping_address_index', implode( ' ', $order->get_address( 'shipping' ) ) );
 		}
 
-		// If customer email changed, update any downloadable permissions.
-		if ( in_array( 'billing_email', $updated_props ) ) {
-			$this->update_downloadable_permissions( $order );
+		// If customer changed, update any downloadable permissions.
+		if ( in_array( 'customer_user', $updated_props ) || in_array( 'billing_email', $updated_props ) ) {
```

Comment:
This should be [customer_id](https://github.com/woocommerce/woocommerce/pull/17895/files#diff-9b0af3108fffa9d52d09b6504882fd0aL262)